File: ui/src/main/java/com/xmartlabs/bigbang/ui/common/recyclerview/SimpleItemRecycleItemType.java
package com.xmartlabs.bigbang.ui.common.recyclerview;

import android.support.annotation.NonNull;

/**
 * {@link RecycleItemType} subclass with an empty {@code onBindViewHolder} implemented.
 *
 * @param <I>  The item related to the {@link android.support.v7.widget.RecyclerView.ViewHolder}.
 * @param <VH> The actual item to be displayed in the {@link android.support.v7.widget.RecyclerView.ViewHolder}.
 */
public abstract class SimpleItemRecycleItemType<I, VH extends SingleItemBaseViewHolder<I>>
    implements RecycleItemType<I, VH> {
  @Override
  public void onBindViewHolder(@NonNull VH viewHolder, @NonNull I item, int position) {

  }
}

File: ui/src/main/java/com/xmartlabs/bigbang/ui/common/recyclerview/SimpleItemRecycleItemType.kt
package com.xmartlabs.bigbang.ui.common.recyclerview

/**
 * [RecycleItemType] subclass with an empty `onBindViewHolder` implemented.

 * @param <I>  The item related to the [android.support.v7.widget.RecyclerView.ViewHolder].
 * *
 * @param <VH> The actual item to be displayed in the [android.support.v7.widget.RecyclerView.ViewHolder].
 */
abstract class SimpleItemRecycleItemType<in I : Any, VH : SingleItemBaseViewHolder<I>> : RecycleItemType<I, VH> {
  override fun onBindViewHolder(viewHolder: VH, item: I, position: Int) { }
}
